File: TwoPointers/1768_Merge_Strings_Alternately/1768_Merge_Strings_Alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder res = new StringBuilder(); // StringBuildser because String concatenation is inefficient and will be Time Limit Exceeded
        
        int n1 = word1.length();
        int n2 = word2.length();

        if(n1==0)   
            res.append(word2);
        else if (n2==0)   
            res.append(word1);
        else {
            int i = 0;
            int j = 0;
            while(i<n1 && j<n2) {
                res.append(word1.charAt(i));
                i++;
                res.append(word2.charAt(j));
                j++;
            }
            if(i>=n1)
                res.append(word2.substring(j));
            if(j>=n2)
                res.append(word1.substring(i));
        }

        return res.toString();
    }
}

File: TwoPointers/1768_Merge_Strings_Alternately/1768_Merge_Strings_Alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string res = "";
        size_t n1 = word1.length();
        size_t n2 = word2.length();

        size_t i = 0, j = 0;
        while(i<n1 && j<n2) {
            res += word1[i++];
            res += word2[j++];
        }
        while(i<n1) {
            res += word1[i++];
        }
        while(j<n2) {
            res += word1[j++];
        }
        
        return res;
    }       
};